meta/saiserialize.cpp

```diff
@@ -2383,6 +2455,45 @@ void sai_deserialize_qos_map_list(
     }
 }
 
+void sai_deserialize_map_list(
+    _In_ const std::string &s,
+    _Out_ sai_map_list_t &maplist,
+    _In_ bool countOnly)
+{
+    SWSS_LOG_ENTER();
+
+    json j = json::parse(s);
+
+    maplist.count = j["count"];
+
+    if (countOnly)
+    {
+        return;
+    }
+
+    if (j["list"] == nullptr)
+    {
+        maplist.list = NULL;
+        return;
+    }
+
+    json arr = j["list"];
+
+    if (arr.size() != (size_t)maplist.count)
+    {
+        SWSS_LOG_THROW("map list count mismatch %lu vs %u", arr.size(), maplist.count);
```

Comment:
I'd change this for an assert (if it's used in sairedis code). When serializing, you'll add only as many items as `maplist.count`, so you should never have more or fewer items in `list` than `maplist.count`

Reply:
I believe it's fine as it stands. This code is in line with other deserialization functions where count is checked.